### ISSUE TITLE
changing coordinate frame of grid overlay

### ISSUE THREAD

**Issue by arusstaylor** [CARTAvis/carta-frontend#538]
Changing the coordinate frame of the grid overlay is too complicated.  It should require only one click.  Currently one must go into the view, selected layouts, select global, select coordinate and then select the coordinate frame.  It should be much easier than this.  The use case for this feature is not that you are customizing overlay settings.  Rather the user wants to be able to dynamically change the grid from equatorial to galactic for instance to check the galactic coordinates of a feature of see how what you are looking at relates to the galaxy.  I suggest we put "coordinate frame" directly in the overlay menu so that is can be changed with one click.

**Comment by veggiesaurus** [comment]
I agree. I think we need to re-think the overlay approach considerably. I see the myriad of options in the complex overlay settings dialog as something that should only be resorted to under two scenarios:
- a user wants to produce a publication-ready plot
- via the scripting interface

Switching between coordinate types should be very easy, and will be especially useful when overlaying contours of a different coordinate type (e.g. overlaying galactic contours on FK5).

@kswang1029 how do you think we should handle this? I also think we should change the nomenclature a bit. Something like coordinate grid overlay?

**Comment by kswang1029** [comment]
How about we borrow the idea from the region button? That is, when the grid button is double clicked, a column of buttons with different coordinate frames shows up (just like regions) so we can switch between different reference frames easier than what we do now.

**Comment by veggiesaurus** [comment]
> How about we borrow the idea from the region button? That is, when the grid button is double clicked, a column of buttons with different coordinate frames shows up (just like regions) so we can switch between different reference frames easier than what we do now.

I like that idea. So something like a button with the following options (tooltip content in brackets):
- Auto (Automatically select the coordinate system based on file headers)
- FK5 (FK5 coordinates, J2000.0 equinox)
- FK4 (FK4 coordinates, B1950.0 equinox)
- GAL (Galactic coordinates)
- ECL (Ecliptic coordinates)
- ICRS (International Celestial Reference System)

**Comment by veggiesaurus** [comment]
@zarda can you have a look at this? it should be relatively simple to add a button group similar to the region button in the image view toolbar, as well as some menu entries for the view menu:

View->Coordinate System->Auto
View->Coordinate System->FK5
...etc

**Comment by veggiesaurus** [comment]
@kswang1029 do we also need entries in the menu?

**Comment by kswang1029** [comment]
Do you mean a dedicated list in the menu? If so I think it is a bit redundant as we already have it in the overlay dialogue. Maybe check with Russ?

**Comment by kswang1029** [comment]
I think it might be good to add a button some where to show the overlay dialogue as a short cut for the menu

**Comment by veggiesaurus** [comment]
> Do you mean a dedicated list in the menu? If so I think it is a bit redundant as we already have it in the overlay dialogue. Maybe check with Russ?

@arusstaylor we now have toolbar selection of coordinates, do we still need menu entries?

@kswang1029 an argument I can make for still including it in the "view" menu is that it corresponds to more than just the image view overlay (spatial profiles, cursor info, region info). Once the spectral profiler's unit conversion is done, we may want to include this as well:

- View->Coordinate Type->Auto, FK5, Galactic, etc
- View->Spectral Axis Type->Auto, Velocity, Frequency, Wavelength, etc
(users would still be able to change the spectral axis for a _specific_ spectral profiler, but would default to the selected axis type)

**Comment by arusstaylor** [comment]
I like this approach for the tool bar. However I think the 'AUTO' is the wrong label for the tool.  It doesn't provide any info about what the tool is. Could we label it something like 'WCS' so it is obvious what it means.

**Comment by veggiesaurus** [comment]
> I like this approach of using the shortcut tool bar. However I think the 'AUTO' is the wrong label for the tool. It doesn't provide any info about what the tool is. Could we label it something like 'WCS' so it is obvious what it means.

That's true. WCS would be an appropriate label.

**Comment by kswang1029** [comment]
yes, ds9 uses "wcs". In addition, ds9 also provides buttons for the coordinate formats. Maybe we can add it to as a separate issue?
![Screen Shot 2019-11-05 at 15 24 59](https://user-images.githubusercontent.com/20819712/68187053-9b040580-ffe0-11e9-8bb3-08c6c6cd66dd.png)

**Comment by veggiesaurus** [comment]
> yes, ds9 uses "wcs". In addition, ds9 also provides buttons for the coordinate formats. Maybe we can add it to as a separate issue?
> ![Screen Shot 2019-11-05 at 15 24 59](https://user-images.githubusercontent.com/20819712/68187053-9b040580-ffe0-11e9-8bb3-08c6c6cd66dd.png)

that could be a separate menu entry, I guess. I think we need to move those sort of things to the menu, rather than the image view toolbar, as they affect multiple widgets

**Comment by arusstaylor** [comment]
I think the tool bar on the image should be viewed as a short cut/convenience tool bar. It should not be a place to set user preferences.  Just a quick way to implement commonly used features via one click.

**Comment by veggiesaurus** [comment]
> I think the tool bar on the image should be viewed as a short cut/convenience tool bar. It should not be a place to set user preferences. Just a quick way to implement commonly used features via one click.

It's displayed in the image viewer, so it should be limited to things that are primarily related to the image viewer. The choice of degrees vs sexagesimal affects spatial profilers, region widgets and quite a bit more.

**Comment by zarda** [comment]
We could add a label (AUTO) following the coordinate system reading from file, like Galactic(AUTO). Let this AUTO to be an action behind to  change the coordinate system after reading a new file or appending any new image.
